sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+log messages when verbose logging is enabled.
+
+### Configuring Log4J 2
+
+1. Add the following dependencies in your pom.xml.
+    ```xml
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-api</artifactId>
+        <version>2.14.1</version>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-core</artifactId>
+        <version>2.14.1</version>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-slf4j-impl</artifactId>
+        <version>2.14.1</version>
+    </dependency>
+    <dependency>
+        <groupId>org.codehaus.groovy</groupId>
+        <artifactId>groovy-jsr223</artifactId>
+        <version>3.0.9</version>
+        <scope>runtime</scope>
+    </dependency>
+    ```
+2. Add [log4j2.xml][log4j2] to your `src/main/resources`.
+
+### Configuring logback
+
+1. Add the following dependencies in your pom.xml.
+    ```xml
+    <dependency>
+        <groupId>ch.qos.logback</groupId>
+        <artifactId>logback-classic</artifactId>
+        <version>1.2.6</version>
+    </dependency>
+    <dependency>
+        <groupId>org.codehaus.janino</groupId>
+        <artifactId>janino</artifactId>
+        <version>3.1.6</version>
+    </dependency>
+    ```
+1. Add [logback.xml][logback] to your `src/main/resources`.
+
+### Enable AMQP transport logging
+
+If enabling client logging is not enough to diagnose your issues.  You can enable logging to a file in the underlying
+AMQP library, [Qpid Proton-J][qpid_proton_j_apache].  Qpid Proton-J uses `java.util.logging`. You can enable logging by
+create a configuration file with the contents below.  Or set `proton.trace.level=ALL` and whichever configuration options
+you want for the `java.util.logging.Handler` implementation.  Implementation classes and their options can be found in
+[Java 8 SDK javadoc][java_8_sdk_javadocs].
+
+To trace the AMQP transport frames, set the environment variable: `PN_TRACE_FRM=1`.
+
+#### Sample "logging.properties" file
+
+The configuration file below logs trace output from proton-j to the file "proton-trace.log".
+
+```
+handlers=java.util.logging.FileHandler
+.level=OFF
+proton.trace.level=ALL
+java.util.logging.FileHandler.level=ALL
+java.util.logging.FileHandler.pattern=proton-trace.log
+java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
+java.util.logging.SimpleFormatter.format=[%1$tF %1$tr] %3$s %4$s: %5$s %n
+```
+
+### Reduce logging
+
+One way to decrease logging is to change the verbosity.  Another is to add filters that exclude logs from logger names like `com.azure.messaging.eventhubs` or `com.azure.core.amqp`.  Examples of this can be found in the xml files in [Configuring Log4J 2](#configuring-log4j-2) and [Configure logback](#configuring-logback).
```

Comment:
What logger names would you recommend a customer leave on if they wanted to submit a bug?